src/Features/Core/Portable/NavigationBar/AbstractNavigationBarItemService.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Remote;
+
+namespace Microsoft.CodeAnalysis.NavigationBar
+{
+    internal abstract class AbstractNavigationBarItemService : INavigationBarItemService
```

Comment:
base type of VB and C# impls.  handles the common logic of trying to call into OOP.